### PR TITLE
Transportation: Push `during` down to `when`. Rename `at` to `between`. Drop reason.

### PR DESCRIPTION
**Pull request by vcschapp**
# Description

*Brief description of the business purpose and effect of the pull request.*

This pull request includes three changes to affecting the transportation schema:

1. The reason property of prohibited transitions is dropped. This is because the reason for a prohibited transition is that it's prohibited; and that the other members of the reason enumeration are physical inhibitors that will be covered by the upcoming inhibitors design.
2. The `at` property used for geometric range scoping is renamed to `between`, and the documentation is updated to explain that `at` is used for single-point scoping whereas `between` is used for ranges.
3. The `during` property used for temporal scoping is pushed down into the `when` clause on rules that support temporal scoping. This is to unblock the implementation of the "except" proposal.

# Reference

*List of relevant links to GitHub issues, PRs, and other documentation.*

1. FIXED: https://github.com/OvertureMaps/tf-transportation/issues/112
2. IMPLEMENTED: https://github.com/OvertureMaps/tf-transportation/issues/98
3. IMPLEMENTED: https://github.com/OvertureMaps/tf-transportation/issues/110
4. UNBLOCKED:  https://github.com/OvertureMaps/tf-transportation/issues/111
5. COMPANION PR WITH COMBOBULATION CHANGES: https://github.com/OvertureMaps/tf-transportation/pull/165

# Testing

*Brief description of the testing done for this change showing why you are confident it works as expected and does not introduce regressions. Provide sample output data where appropriate.* 

Updated existing examples and counterexamples.

# Checklist

*Checklist of tasks commonly-associated with schema pull requests. Please review the relevant checklists and ensure you do all the tasks that are required for the change you made.*

1. [ ] Add relevant examples.
6. [ ] Add relevant counterexamples.
7. [x] Update in-schema documentation using plain English written in complete sentences, if an update is required.
8. [ ] Update Docusaurus documentation, if an update is required.
9. [ ] Review change with Overture technical writer to ensure any advanced documentation needs will be taken care of, unless the change is trivial and would not affect the documentation.

# Documentation Website

*Update the hyperlink below to put the pull request number in.*

[Docs preview for this PR.](https://dfhx9f55j8eg5.cloudfront.net/pr/151)

### CODE REVIEW

**Comment by vcschapp** [comment]
Tagging @jwass, @jenningsanderson, and @DavidKarlas from Schema Task Force because one effect of this PR is that it slightly changes the `names` schema because, in `rules` objects, the property formerly known as `"at"` is renamed to `"between`". This means admins, base, buildings, divisions, and places themes need to update their Parquet schema definition.

**Review by ylinzhi — APPROVED** [review]
𓊝 𓊝 𓊝

File: examples/transportation/segment/road/lanes/lanes-flow-reversible.yaml [hunk 1]
```diff
@@ -25,8 +25,8 @@ properties:
               - access_type: allowed
                 when:
                   heading: forward
-                during: Mo-Su 00:00-12:00
+                  during: Mo-Su 00:00-12:00
```
Comment (brad-richardson) [inline examples/transportation/segment/road/lanes/lanes-flow-reversible.yaml:28]: I don't love `during` here, it reads weird: "when heading is" vs "when during is". Would it make sense to rename this to "time"?

**Review by brad-richardson — APPROVED** [review]

**Review by bastiaanvanassche-tomtom — APPROVED** [review]

**Review by sanjibdutta1 — APPROVED** [review]
LGTM. Thanks for doing this!

**Pull request merged**